src/metabase/models/database.clj

```diff
@@ -152,14 +152,20 @@
   "**MetabasePass**")
 
 ;; when encoding a Database as JSON remove the `details` for any non-admin User. For admin users they can still see
-;; the `details` but remove the password. No one gets to see this in an API response!
+;; the `details` but remove anything resembling a password. No one gets to see this in an API response!
 (add-encoder
  DatabaseInstance
  (fn [db json-generator]
    (encode-map
-    (cond
-      (not (:is_superuser @*current-user*)) (dissoc db :details)
-      (get-in db [:details :password])      (assoc-in db [:details :password] protected-password)
-      (get-in db [:details :pass])          (assoc-in db [:details :pass] protected-password) ; MongoDB uses "pass" instead of password
-      :else                                 db)
+    (cond-> db
+      (get-in db [:details :password])                      (assoc-in [:details :password] protected-password)
+      (get-in db [:details :pass])                          (assoc-in [:details :pass] protected-password) ; MongoDB uses "pass" instead of password
+      (get-in db [:details :tunnel-pass])                   (assoc-in [:details :tunnel-pass] protected-password)
+      (get-in db [:details :tunnel-private-key])            (assoc-in [:details :tunnel-private-key] protected-password)
+      (get-in db [:details :tunnel-private-key-passphrase]) (assoc-in [:details :tunnel-private-key-passphrase] protected-password)
+      (get-in db [:details :access-token])                  (assoc-in [:details :access-token] protected-password)
+      (get-in db [:details :refresh-token])                 (assoc-in [:details :refresh-token] protected-password)
+      (get-in db [:details :service-account-json])          (assoc-in [:details :service-account-json] protected-password)
```

Comment:
this would probably be a lot cleaner if you changed it to something like

```clj
(update db :details (fn [details]
                       (reduce
                        #(m/update-existing %1 %2 protected-password)
                        details
                        [:password :pass :tunnel-pass ...])))
```

You can use [Medley `update-existing`](https://weavejester.github.io/medley/medley.core.html#var-update-existing)

Reply:
ahhh cool I can change that... I knew there had to be something easier than all the duplicates in there